### PR TITLE
Lock Free List - WIP

### PR DESCRIPTION
**Pull request by tsvisabo**
* Add CAS (compare and swap impl)

### CODE REVIEW

**Comment by jserv** [comment]
@jkrvivian, lock-free linked list implementation is an area where you can contribute.

File: utils/handles/cas.h [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+#ifndef __UTILS_HANDLES_CAS_H__
+#define __UTILS_HANDLES_CAS_H__
+
+#include <stdbool.h>
+
+#if !defined(_WIN32) && defined(__unix__) || defined(__unix) || \
+    (defined(__APPLE__) && defined(__MACH__))
+#include <unistd.h>
+#elif defined(_WIN32)
+#include <Windows.h>
```
Comment (jserv) [inline utils/handles/cas.h:17]: Add extra `#error` statement if no supported environment is found.

File: utils/handles/cas.h [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    return true;
+  }
+  return false;
+}
+#elif __ENVIRONMENT_MAC_OS_X_VERSION_MIN_REQUIRED__ >= 1050
+static inline bool compare_and_swap(volatile void** ptr, void* comp_val,
+                                    void* new_val) {
+  return OSAtomicCompareAndSwapPtr(comp_val, new_val, ptr);
+}
+
```
Comment (jserv) [inline utils/handles/cas.h:42]: Add `#else` here.

**Comment by th0br0** [comment]
closing in favour of existing liblfds impls

**Pull request closed without merging**